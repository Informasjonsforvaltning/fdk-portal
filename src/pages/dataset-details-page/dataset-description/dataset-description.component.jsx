import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import sanitizeHtml from 'sanitize-html';

import localization from '../../../lib/localization';
import { ShowMore } from '../../../components/show-more/show-more';
import { SearchHitHeader } from '../../../components/search-hit-header/search-hit-header.component';
import { getTranslateText } from '../../../lib/translateText';

export class DatasetDescription extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showAll: false
    };
    this.toggleShowAll = this.toggleShowAll.bind(this);
  }

  toggleShowAll() {
    const { showAll } = this.state;
    this.setState({ showAll: !showAll });
  }

  render() {
    const { datasetItem, referenceData } = this.props;
    const { publisher, theme, provenance, catalog } = datasetItem || {};
    let { title, descriptionFormatted, objective } = datasetItem || {};
    title = getTranslateText(title);
    descriptionFormatted = getTranslateText(descriptionFormatted);
    objective = getTranslateText(objective);

    return (
      <header>
        <SearchHitHeader
          title={title}
          publisherLabel={localization.search_hit.owned}
          publisher={publisher}
          catalog={catalog}
          theme={theme}
          nationalComponent={_.get(provenance, 'code') === 'NASJONAL'}
          referenceData={referenceData}
          darkThemeBackground
        />

        {descriptionFormatted && (
          <ShowMore showMoreButtonText={localization.showFullDescription}>
            <span
              dangerouslySetInnerHTML={{
                __html: sanitizeHtml(descriptionFormatted)
              }}
            />
          </ShowMore>
        )}

        {objective && (
          <ShowMore showMoreButtonText={localization.showFullObjective}>
            <strong>{localization.objective}: </strong>
            {objective}
          </ShowMore>
        )}
      </header>
    );
  }
}

DatasetDescription.defaultProps = {
  datasetItem: null,
  referenceData: null
};

DatasetDescription.propTypes = {
  datasetItem: PropTypes.object,
  referenceData: PropTypes.object
};
