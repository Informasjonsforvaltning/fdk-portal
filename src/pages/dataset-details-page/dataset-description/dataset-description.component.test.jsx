import React from 'react';
import { shallow } from 'enzyme';
import { DatasetDescription } from './dataset-description.component';
import dataset from '../__fixtures/datasetApiResponse.json';

test('should render DatasetDescription correctly with no props', () => {
  const minWrapper = shallow(<DatasetDescription />);
  expect(minWrapper).toMatchSnapshot();
});

test('should render DatasetDescription correctly', () => {
  const defaultProps = {
    datasetItem: dataset
  };
  const wrapper = shallow(<DatasetDescription {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
