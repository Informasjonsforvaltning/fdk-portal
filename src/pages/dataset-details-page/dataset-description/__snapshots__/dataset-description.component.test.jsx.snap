// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DatasetDescription correctly 1`] = `
<header>
  <SearchHitHeader
    catalog={
      Object {
        "description": Object {
          "nb": "GeoNorge er den nasjonale katalogen for geografisk informasjon",
        },
        "issued": "2018-02-13T00:00:00+0100",
        "language": "http://publications.europa.eu/resource/authority/language/NOR",
        "modified": "2018-01-19T00:00:00+0100",
        "publisher": Object {
          "id": "971040238",
          "naeringskode": Object {
            "code": "71.123",
            "prefLabel": Object {
              "no": "Kart og oppmåling",
            },
            "uri": "http://www.ssb.no/nace/sn2007/71.123",
          },
          "name": "STATENS KARTVERK",
          "orgPath": "/STAT/972417858/971040238",
          "organisasjonsform": "ORGL",
          "overordnetEnhet": "972417858",
          "prefLabel": Object {
            "no": "Kartverket",
          },
          "sektorkode": Object {
            "code": "1110",
            "prefLabel": Object {
              "no": "Statens forretningsdrift",
            },
            "uri": "http://www.brreg.no/sektorkode/1110",
          },
          "type": "no.dcat.datastore.domain.dcat.Publisher",
          "uri": "http://data.brreg.no/enhetsregisteret/enhet/971040238",
          "valid": true,
        },
        "themeTaxonomy": Array [
          "https://register.geonorge.no/subregister/metadata-kodelister/kartverket/nasjonal-temainndeling",
          "http://publications.europa.eu/mdr/resource/authority/data-theme/html/data-theme-eng.html",
          "http://www.eionet.europa.eu/gemet/inspire_themes",
        ],
        "title": Object {
          "nb": "Geonorge",
        },
        "uri": "http://www.geonorge.no/geonetwork",
      }
    }
    darkThemeBackground={true}
    externalLink={false}
    nationalComponent={false}
    publisher={
      Object {
        "id": "971040238",
        "naeringskode": Object {
          "code": "71.123",
          "prefLabel": Object {
            "no": "Kart og oppmåling",
          },
          "uri": "http://www.ssb.no/nace/sn2007/71.123",
        },
        "name": "STATENS KARTVERK",
        "orgPath": "/STAT/972417858/971040238",
        "organisasjonsform": "ORGL",
        "overordnetEnhet": "972417858",
        "prefLabel": Object {
          "no": "Kartverket",
        },
        "sektorkode": Object {
          "code": "1110",
          "prefLabel": Object {
            "no": "Statens forretningsdrift",
          },
          "uri": "http://www.brreg.no/sektorkode/1110",
        },
        "type": "no.dcat.datastore.domain.dcat.Publisher",
        "uri": "http://data.brreg.no/enhetsregisteret/enhet/971040238",
        "valid": true,
      }
    }
    publisherItems={null}
    publisherLabel="Owner: "
    publisherTag="span"
    referenceData={null}
    statusCode={null}
    tag="h1"
    theme={
      Array [
        Object {
          "code": "GOVE",
          "conceptSchema": Object {
            "id": "http://publications.europa.eu/resource/authority/data-theme",
            "title": Object {
              "en": "Dataset types Named Authority List",
            },
            "versioninfo": "20160921-0",
            "versionnumber": "20160921-0",
          },
          "id": "http://publications.europa.eu/resource/authority/data-theme/GOVE",
          "startUse": "2015-10-01",
          "title": Object {
            "bg": "Правителство и публичен сектор",
            "cs": "Vláda a veřejný sektor",
            "da": "Regeringen og den offentlige sektor",
            "de": "Regierung und öffentlicher Sektor",
            "el": "Κυβέρνηση και δημόσιος τομέας",
            "en": "Government and public sector",
            "es": "Gobierno y sector público",
            "et": "Valitsus ja avalik sektor",
            "fi": "Valtioneuvosto ja julkinen sektori",
            "fr": "Gouvernement et secteur public",
            "ga": "Rialtas agus earnáil phoiblí",
            "hr": "Vlada i javni sektor",
            "hu": "Kormányzat és közszféra",
            "it": "Governo e settore pubblico",
            "lt": "Vyriausybė ir viešasis sektorius",
            "lv": "Valdība un sabiedriskais sektors",
            "mt": "Gvern u settur pubbliku",
            "nb": "Forvaltning og offentlig sektor",
            "nl": "Overheid en publieke sector",
            "pl": "Rząd i sektor publiczny",
            "pt": "Governo e setor público",
            "ro": "Guvern şi sector public",
            "sk": "Vláda a verejný sektor",
            "sl": "Vlada in javni sektor",
            "sv": "Regeringen och den offentliga sektorn",
          },
        },
      ]
    }
    title="Markagrensen"
    titleLink={null}
  />
  <withStateHandlers(ShowMorePure)
    showMoreButtonText="Show the full description"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "Datasettet avgrenser område for virkeområdet til lov 6. juni 2009 nr. 35 om naturområder i Oslo og nærliggende kommuner (markaloven) som trådte i kraft 1. september 2009. Markalovens virkeområde er fastsatt i forskrift 4. september 2015 nr. 1032 om justering av markagrensen fastlegger markalovens geografiske virkeområde med tilhørende kart.",
        }
      }
    />
  </withStateHandlers(ShowMorePure)>
</header>
`;

exports[`should render DatasetDescription correctly with no props 1`] = `
<header>
  <SearchHitHeader
    darkThemeBackground={true}
    externalLink={false}
    nationalComponent={false}
    publisher={null}
    publisherItems={null}
    publisherLabel="Owner: "
    publisherTag="span"
    referenceData={null}
    statusCode={null}
    tag="h1"
    theme={Array []}
    title={null}
    titleLink={null}
  />
</header>
`;
